external-api/src/v1/controllers/number-generator.controller.ts

```diff
@@ -1,85 +1,92 @@
+};
 
-    return {
-      status: response.status || 500,
-    };
+/**
+ * Determines the number type based on the entity type.
+ * @param entityType - The type of entity for which the number type needs to be determined.
+ * @returns A number based on the value of entityType.
+ * @throws {Error} If entityType is invalid.
+ */
+export const getNumberTypeId = (entityType: string): number => {
+  switch (entityType) {
+    case ENTITY_TYPE.DEAL:
+      return NUMBER_TYPE.DEAL;
+    case ENTITY_TYPE.FACILITY:
+      return NUMBER_TYPE.FACILITY;
+    default:
+      throw new InvalidEntityTypeError(entityType);
   }
+};
+
+/**
+ * Retrieves a number from a number generator API based on the provided `entityType` and `dealId`.
+ * @param req - The HTTP request object containing the `entityType` and `dealId` in the `body` property.
+ * @param res - The HTTP response object used to send the response back to the client.
+ * @returns {Promise<Object>} The retrieved number in the response body.
+ */
+export const get = async (req: Request, res: Response): Promise<Response<NumberGeneratorResponse> | Response<NumberGeneratorErrorResponse>> => {
+  try {
+    const { entityType, dealId } = req.body;
+    const numberTypeId = getNumberTypeId(entityType);
+    const endpoint = `${APIM_MDM_URL}numbers`;
+    const payload = {
+      numberTypeId,
+      createdBy: USER.DTFS,
+      requestingSystem: USER.DTFS,
+    };
 
-  const { id: instanceId, ...numberGeneratorFunctionUrls } = response.data;
+    console.info('⚡️ Invoking number generator for deal %s', dealId);
 
-  await durableFunctionsLogController.addDurableFunctionLog({
-    type: 'NUMBER_GENERATOR',
-    data: {
-      instanceId,
-      dealType,
-      entityType,
-      entityId,
-      dealId,
-      numberGeneratorFunctionUrls,
-      user,
-    },
-  });
+    const response: NumberGeneratorResponse = await axios.post(endpoint, [payload], headers);
 
-  return response;
-};
+    if (!response.data) {
+      console.error('❌ Void number generator response received for deal %s %o', dealId, response);
+      throw new Error(`Void number generator response received for deal ${dealId}`, { cause: 'Void response from APIM MDM' });
+    }
+
+    const { status, data } = response;
+
+    if (!data.length) {
+      throw new Error(`Empty number generator response received for deal ${dealId}`, { cause: 'Empty response from APIM MDM' });
+    }
 
-export const callNumberGeneratorPOST = async (req: Request, res: Response) => {
-  const { dealType, entityType, entityId, dealId, user } = req.body;
+    const { maskedId: ukefId } = data[0];
 
-  console.info('External API - calling Number Generator');
+    console.info('✅ UKEF ID received %d for deal %s', ukefId, dealId);
 
-  const { status } = await callNumberGenerator({
-    dealType,
-    entityType,
-    entityId,
-    dealId,
-    user,
-  });
+    return res.status(HttpStatusCode.Ok).send({
+      status,
+      data,
+    });
+  } catch (error: any) {
```

Comment:
I'm surprised :any isn't throwing an eslint failure! 
I think as it's only use or axios throwing errors we should be able to type this as : Error as everything should inherit from the base Error class